Update development/standards guide (#7981)
readthedocs/readthedocs.org: docs/development/standards.rst

#. install `Docker <https://www.docker.com/>`_ following `their installation guide <https://docs.docker.com/install/>`_.

#. clone the ``readthedocs.org`` repository:

   .. prompt:: bash

>       git clone --recurse-submodules git@github.com:readthedocs/readthedocs.org.git

Reviewer: Using a https url is more easy for users since they don't need to setup a ssh key

>       git clone --recurse-submodules git@github.com:readthedocs/readthedocs.org.git

Author: ```suggestion
      git clone --recurse-submodules https://github.com/readthedocs/readthedocs.org/
```

       you can run ``inv docker.attach web`` and jump into a pdb session
       (it also works with ipdb and pdb++)

    .. tip::

>        You can hit CTRL-p CTRL-p to detach it without stopping the running process.

Reviewer 2: Huh, I didn't know that! :)

Author: It's kind of new: https://github.com/readthedocs/common/pull/81

We work on the shadows :smile:

Adding a new Python dependency
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The Docker image for the servers is built with the requirements defined in the checked out branch.

Reviewer: ```suggestion
The Docker image for the servers is built with the requirements defined in the current branch.
```

Reviewer 2: I think both work, could even say `currently checked out branch` :D

> The Docker image for the servers is built with the requirements defined in the checked out branch.

Author: ```suggestion
The Docker image for the servers is built with the requirements defined in the current checked out branch.
```
